File: backend/src/main/java/cz/forgottenempire/servermanager/installation/ServerInstallationDto.java
package cz.forgottenempire.servermanager.installation;

import cz.forgottenempire.servermanager.common.InstallationStatus;
import cz.forgottenempire.servermanager.common.ServerType;
import cz.forgottenempire.servermanager.steamcmd.ErrorStatus;
import lombok.Builder;
import lombok.Data;

import java.util.Set;

@Data
@Builder
class ServerInstallationDto {

    private ServerType type;
    private String version;
    private InstallationStatus installationStatus;
    private ErrorStatus errorStatus;
    private String lastUpdatedAt;
    private ServerInstallation.Branch branch;
    private Set<ServerInstallation.Branch> availableBranches;
}

File: frontend/src/dtos/ServerInstallationDto.ts
import {ServerType} from "./ServerDto.ts";

export interface ServerInstallationsDto {
    serverInstallations: Array<ServerInstallationDto>
}

export interface ServerInstallationDto {
    type: ServerType,
    version: string,
    installationStatus: string,
    errorStatus: string | null,
    lastUpdatedAt: string,
    branch: string,
    availableBranches: string[]
}
